pkg/tools/syncmap/gen.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncmap
+
+import "sync"
+
+//go:generate go-syncmap -output ns_sync_map.gen.go -type StringRegistryNetworkServiceMap<string,*github.com/networkservicemesh/api/pkg/api/registry.NetworkService>
```

Comment:
Two questions:
1. Why the very long and complicated names?
2. Why put this in a separate package?

Reply:
1. Will fix.
2. Could you please explain when we should put things to `pkg/tools`?

Reply:
@denis-tingajkin Still feeling that out.   Here's are some of the dynamic things that I tend to think about balancing:

1.  Is there a real (not imagined) second use for the code?
2. How much work is it to correctly reproduce the code locally in that second use?

The thinking is that every hop of indirection produces some obscuring of what is happening.

So in this example, there's basically a single //go:generate line to reproduce a map of this sort... super small, best to keep the code local.

Reply:
ok, got it, thanks!